honestcash-v2/src/app/core/guards/unauthorized.guard.ts

```diff
@@ -1,25 +1,23 @@
-import { Injectable } from '@angular/core';
-import { Router, CanActivate } from '@angular/router';
+import {Injectable} from '@angular/core';
+import {CanActivate, Router} from '@angular/router';
 
-import { Logger } from '../services/logger.service';
-import { AuthenticationService } from '../services/authentication.service';
+import {Logger} from '../services/logger.service';
+import {AuthService} from '../services/auth.service';
 
 const log = new Logger('UnauthorizedGuard');
 
 @Injectable({providedIn: 'root'})
 export class UnauthorizedGuard implements CanActivate {
-  constructor(private router: Router, private authenticationService: AuthenticationService) {}
+  constructor(private router: Router, private authenticationService: AuthService) {
+  }
 
   canActivate(): boolean {
     if (!this.authenticationService.hasAuthorization()) {
       return true;
     }
 
     log.debug('Authorized, redirecting to feed page...');
-
-    // @todo - v2 migration (this redirects to v1 with a hard refresh)
-    location.href = '/';
-    // this.router.navigateByUrl('/feed');
+    this.router.navigateByUrl('/feed');
```

Comment:
VersionOneGuard kicks in here so we can safely keep using this guard with its original functionality. VersionOneGuard is specified before other guards so it is always run first

Reply:
good idea.